.github/workflows/build.yml

```diff
@@ -0,0 +1,162 @@
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  TinyCBOR:
+    timeout-minutes: 45
+    # Common environment variables
+    env:
+      HOMEBREW_NO_INSTALL_CLEANUP: 1
+      HOMEBREW_NO_ANALYTICS: 1
+
+    strategy:
+      # Always run all jobs in the matrix, even if one fails.
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        build_cfg: [
+          { "name": "gcc-no-math",
+            "flags":
+              '{ "QMAKESPEC": "linux-gcc-no-math",
+                 "EVAL": "export CXX=false && touch src/math.h src/float.h",
+                 "CFLAGS": "-ffreestanding -DCBOR_NO_FLOATING_POINT -Os",
+                 "LDFLAGS": "-Wl,--no-undefined",
+                 "LDLIBS": ""
+              }',
+          },
+          { "name": "gcc-freestanding",
+            "flags":
+              '{ "QMAKESPEC": "linux-gcc-freestanding",
+                 "EVAL": "export CXX=false",
+                 "CFLAGS": "-ffreestanding -Os",
+                 "LDFLAGS": "-Wl,--no-undefined -lm"
+              }',
+          },
+          { "name": "clang",
+            "flags":
+              '{ "QMAKESPEC": "linux-clang",
+                 "EVAL": "export CC=clang && export CXX=clang++",
+                 "CFLAGS": "-Oz",
+                 "LDFLAGS": "-Wl,--no-undefined -lm",
+                 "QMAKEFLAGS": "-config release",
+                 "MAKEFLAGS": "-s",
+                 "TESTARGS": "-silent"
+              }',
+          },
+          { "name": "linux-g++",
+            "flags":
+              '{ "QMAKESPEC": "linux-g++",
+                 "EVAL": "export CC=gcc && export CXX=g++",
+                 "CFLAGS": "-Os",
+                 "LDFLAGS": "-Wl,--no-undefined -lm",
+                 "QMAKEFLAGS": "-config release",
+                 "QT_NO_CPU_FEATURE": "rdrnd"
+              }'
+          }
+        ]
+        include:
+          - os: macos-11
+            build_cfg: { "name": "clang",
+                         "flags":
+                           '{ "QMAKESPEC": "macx-clang",
+                              "CFLAGS": "-Oz",
+                              "QMAKEFLAGS": "-config debug",
+                              "MAKEFLAGS": "-s",
+                              "TESTARGS": "-silent",
+                              "PATH": "/usr/local/opt/qt5/bin:$PATH"
+                            }'
+                        }
+
+    # Default job name is too long to be visible in the "Checks" tab.
+    name: ${{ matrix.os }}/${{ matrix.build_cfg.name }}
+    # The type of runner that the job will run on
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Clone tinycbor
+      uses: actions/checkout@v4
+
+    - name: install Linux software
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        # Need a recent Valgrind, otherwise debug info cannot be read.
+        sudo snap install valgrind --classic
```

Comment:
And "ubuntu-latest"'s valgrind package using APT is too old? Ugh.

Reply:
ubuntu-latest is 22.04 LTS, so it's a Valgrind 3.17 in the distribution (or something like that).